src/ast_fwd_decl.hpp

```diff
@@ -187,72 +199,176 @@ namespace Sass {
   IMPL_MEM_OBJ(Supports_Interpolation);
   IMPL_MEM_OBJ(At_Root_Query);
   IMPL_MEM_OBJ(Null);
-  IMPL_MEM_OBJ(Parent_Selector);
   IMPL_MEM_OBJ(Parent_Reference);
   IMPL_MEM_OBJ(Parameter);
   IMPL_MEM_OBJ(Parameters);
   IMPL_MEM_OBJ(Argument);
   IMPL_MEM_OBJ(Arguments);
   IMPL_MEM_OBJ(Selector);
   IMPL_MEM_OBJ(Selector_Schema);
-  IMPL_MEM_OBJ(Simple_Selector);
+  IMPL_MEM_OBJ(SimpleSelector);
   IMPL_MEM_OBJ(Placeholder_Selector);
   IMPL_MEM_OBJ(Type_Selector);
   IMPL_MEM_OBJ(Class_Selector);
   IMPL_MEM_OBJ(Id_Selector);
   IMPL_MEM_OBJ(Attribute_Selector);
   IMPL_MEM_OBJ(Pseudo_Selector);
-  IMPL_MEM_OBJ(Wrapped_Selector);
-  IMPL_MEM_OBJ(Compound_Selector);
-  IMPL_MEM_OBJ(Complex_Selector);
-  IMPL_MEM_OBJ(Selector_List);
+  
+
+  IMPL_MEM_OBJ(CompoundOrCombinator);
+  IMPL_MEM_OBJ(SelectorCombinator);
+  IMPL_MEM_OBJ(CompoundSelector);
+  IMPL_MEM_OBJ(ComplexSelector);
+  IMPL_MEM_OBJ(SelectorList);
 
   // ###########################################################################
-  // Implement compare, order and hashing operations for AST Nodes
+  // Implement compare and hashing operations for raw pointers
   // ###########################################################################
 
-  struct HashNodes {
+  template <class T>
+  size_t PtrHashFn(const T* ptr) {
+    return std::hash<std::size_t>()((size_t)ptr);
+  }
+
+  struct PtrHash {
+    template <class T>
+    size_t operator() (const T* ptr) const {
+      return PtrHashFn(ptr);
+    }
+  };
+
+  template <class T>
+  bool PtrEqualityFn(const T* lhs, const T* rhs) {
```

Comment:
Somehwat true, but still useful to explicitly give the compare function for map structures.
That way it becomes very obvious/readable by what a map is hashed or compared.

Reply:
I think it's the opposite of obvious, because the name `PtrEqualityFn` is ambiguous (are the pointers compared, or the objects they point to?).

Whereas the default comparator is actually obvious.